src/slacko.ml

```diff
@@ -765,33 +765,22 @@ let groups_list ?exclude_archived session =
       (match d |> groups_list_obj_of_yojson with
         | Result.Ok x -> `Success x.groups
         | Result.Error x -> `ParseFailure x)
-    | #bot_error
     | #parsed_auth_error as res -> res
     | _ -> `Unknown_error
 
-exception No_matches
-exception No_unique_matches
-exception Lookup_failed
-
 (* look up the id of query from results provided by the listfn *)
 let lookupk session listfn filterfn k =
   match%lwt listfn session with
-  | `Success channels -> (match List.filter filterfn channels with
-    | [] -> Lwt.fail No_matches
-    | [x] -> Lwt.return @@ k x
-    | _ -> Lwt.fail No_unique_matches)
-  | _ -> Lwt.fail Lookup_failed
+  | #parsed_auth_error as e -> Lwt.return e
+  | `Success items -> Lwt.return @@ k @@ List.filter filterfn items
 
-let id_of_channel session : _ -> [ `User_not_found | `Channel_not_found | `Found of string ] Lwt.t = function
+let id_of_channel session = function
   | ChannelId id -> Lwt.return @@ `Found id
   | ChannelName name ->
     let base = String.sub name 1 @@ String.length name - 1 in
-    match%lwt (lookupk session channels_list (fun (x:channel_obj) -> x.name = base) @@ function
-    | {id = ChannelId s; _} -> Some s
-    | {id = ChannelName _; _} -> None) with
-    | exception _ -> Lwt.return `Channel_not_found
-    | Some v -> Lwt.return @@ `Found v
-    | None -> Lwt.return `Channel_not_found
+    lookupk session channels_list (fun (x:channel_obj) -> x.name = base) @@ function
+    | [{id = ChannelId s; _}] -> `Found s
```

Comment:
The match won't blow up, because the `` | _ -> `Channel_not_found `` case will catch any list that doesn't contain exactly one item.